### ISSUE TITLE
Heuristic for textual file recognition

### ISSUE THREAD

**Issue by JanKrivanek** [VerifyTests/Verify#742]
### Request

Add logic for differentiation binary and textual files based on their content
This might possibly be opt-in, as it'd slow down the processing a bit.

### Motivation

Currently the textual files are recognized only by their extension. So there can be snapshots which are considered non-textual despite they can be handled as texts.
Textual comparison is preferable to stream comparison for multiple reasons (ability to normalize differences in conventions like newlines, BOMs, etc.; ability to provide more actionable equality failures). So extra efforts in trying to differentiate binary and textual files might bring extra benefit to users

### Possible implementation

Unfortunately there is no easy solution :-) Probably someting along the lines of taking a sample of file and doing frequency analysis of bytes/group of bytes and looking for high presence of nonprintable characters.
Existing third party libraries exist: https://github.com/tylerlong/ude, https://github.com/superstrom/chardetsharp etc.

**Comment by SimonCropp** [comment]
wouldn’t this suffice ? https://github.com/VerifyTests/EmptyFiles#addtextextension

**Comment by SimonCropp** [comment]
@JanKrivanek ?

**Comment by JanKrivanek** [comment]
Thank you @SimonCropp - I wasn't aware of the `addextension` - thinking about it bit more it should likely suffice - as heuristic would anyway need to be opt in - so than why not just opt in extensions. In my case I was thinking about a scenario where `Verify` is used by an intermediate tooling, where final client code doesn't have access to settings - but the extension choices can be simply passed through, or if needed the heuristic can live in the intermediate tooling and then simply just call `addextension` based on results.

**tl;dr;:** closing this (can be reopened if there are more asks for it.)
